include/libtorrent/config.hpp

```diff
@@ -56,6 +57,14 @@ POSSIBILITY OF SUCH DAMAGE.
 #pragma GCC diagnostic ignored "-Wformat-extra-args"
 #endif
 
+// ====== BOOST ========
+
+#if BOOST_VERSION >= 107000
+#define GET_IO_SERVICE(s) ((boost::asio::io_context&)(s).get_executor().context())
```

Comment:
I would rather not use a macro for this. Also, I'm not a fan of the reinterpret cast. Is there a reason calling `get_executor()` doesn't work without the cast?
Also, I think the `get_executor()` call dates back quite far, to boost-1.66 or so. So I *think* it would be reasonable to just require a new version of boost and replace all calls to `get_io_service()` with `get_executor().context()` instead.

I made this change in `master`, and I think it should be done in the same way in `RC_1_2`, to make the merge simpler. (i.e. merge of `RC_1_2` into `master`)

Reply:
If you think it's better to have an abstraction like this, to support older versions of boost, please make it an inline function instead

Reply:
OK, I can remove the fallback and bump boost requirement from 1.54 to 1.66.
Concerning the reinterpret_cast, without it, build fails on: 
```
../include/libtorrent/proxy_base.hpp: In member function ‘libtorrent::io_service& libtorrent::proxy_base::get_io_service()’:
../include/libtorrent/proxy_base.hpp:249:39: error: invalid initialization of reference of type ‘libtorrent::io_service& {aka boost::asio::io_context&}’ from expression of type ‘boost::asio::execution_context’
   return m_sock.get_executor().context();
```